### PR TITLE
[SQL-129] Update H2 from v1.4.200 to v2.0.202

### PR DESCRIPTION
**Pull request by kelvinqian00**
Update since v1.4.200 is no longer supported and to address CVE-2021-23463.

**NOTE:** This update is **not** backwards compatible (if using H2) due to changes in SQL syntax and to the underlying database file.

### CODE REVIEW

File: src/db/h2/lrsql/h2/sql/ddl.sql [hunk 1]
```diff
@@ -156,6 +156,7 @@ CREATE TABLE IF NOT EXISTS lrs_credential (
   api_key    VARCHAR(255) NOT NULL,
   secret_key VARCHAR(255) NOT NULL,
   account_id UUID NOT NULL,
+  UNIQUE (api_key, secret_key),
```
Comment (cliffcaseyyet) [inline src/db/h2/lrsql/h2/sql/ddl.sql:159]: was this related to upgrade or just another unassociated improvement?

File: src/db/h2/lrsql/h2/sql/ddl.sql [hunk 2]
```diff
@@ -117,7 +117,7 @@ CREATE TABLE IF NOT EXISTS agent_profile_document (
   last_modified  TIMESTAMP NOT NULL,
   content_type   VARCHAR(255) NOT NULL,
   content_length INTEGER NOT NULL,
-  contents       BINARY NOT NULL,
+  contents       BINARY VARYING NOT NULL,
```
Comment (cliffcaseyyet) [inline src/db/h2/lrsql/h2/sql/ddl.sql:120]: i read up the difference. is this just to account for not declaring size?

**Review by cliffcaseyyet — APPROVED** [review]

File: src/db/h2/lrsql/h2/sql/ddl.sql (hunk 1 above)
Comment (kelvinqian00) [reply src/db/h2/lrsql/h2/sql/ddl.sql:159]: This is upgrade related; when not present, this error occurs:
```
11:38:34.894 [main] INFO  lrsql.system.database - Starting new connection for h2:mem database...
Execution error (JdbcSQLSyntaxErrorException) at org.h2.message.DbException/getJdbcSQLException (DbException.java:632).
Constraint "PRIMARY KEY | UNIQUE (API_KEY, SECRET_KEY)" not found; SQL statement:
CREATE TABLE IF NOT EXISTS credential_to_scope (
  id         UUID NOT NULL PRIMARY KEY,
  api_key    VARCHAR(255) NOT NULL,
  secret_key VARCHAR(255) NOT NULL,
  scope      ENUM('statements/write',
                  'statements/read',
                  'all/read',
                  'all'),   FOREIGN KEY (api_key, secret_key)
    REFERENCES lrs_credential(api_key, secret_key)
    ON DELETE CASCADE
) [90057-202]
```

File: src/db/h2/lrsql/h2/sql/ddl.sql (hunk 2 above)
Comment (kelvinqian00) [reply src/db/h2/lrsql/h2/sql/ddl.sql:120]: Yes - `BINARY` only works for fixed sized arrays.

**Pull request merged**